plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -306,9 +314,13 @@ def deploy(release_docs)
 
     def deploy_to_cluster(release, deploys)
       deploy(deploys)
-      success = wait_for_resources_to_complete(release, deploys)
-      show_failure_cause(release) unless success
-      success
+      successful = wait_for_resources_to_complete(release, deploys)
+      unless successful
+        rollback(deploys)
+        show_failure_cause(release)
```

Comment:
better show the failure cause first, otherwise if something hangs we will never know